pkg/resourcemonitor/noderesourcesaggregator.go

```diff
@@ -0,0 +1,243 @@
+	if err != nil {
+		return nil, fmt.Errorf("Can't receive response: %v.Get(_) = _, %v", podResourceClient, err)
+	}
+
+	allDevs := GetContainerDevicesFromAllocatableResources(resp, cpuInfo)
+
+	nodeResourceInstance := nodeResources{
+		cpuInfo:           cpuInfo,
+		numaInfo:          &nodes,
+		numaDists:         dists,
+		resourceID2NUMAID: makeResourceMap(len(nodes.Online), allDevs),
+		perNUMACapacity:   MakeNodeCapacity(allDevs),
+	}
+
+	return &nodeResourceInstance, nil
+}
+
+func (nodeResourceData *nodeResources) Aggregate(podResData []PodResources) map[string]*v1alpha1.Zone {
+	perNuma := make(map[int]map[v1.ResourceName]*ResourceData)
+	for nodeNum, nodeRes := range nodeResourceData.perNUMACapacity {
+		perNuma[nodeNum] = make(map[v1.ResourceName]*ResourceData)
+		for resName, resCap := range nodeRes {
+			perNuma[nodeNum][resName] = &ResourceData{
+				capacity:    resCap,
+				allocatable: resCap,
+			}
+		}
+	}
+
+	for _, podRes := range podResData {
+		for _, contRes := range podRes.Containers {
+			for _, res := range contRes.Resources {
+				nodeResourceData.updateAllocatable(perNuma, res)
+			}
+		}
+	}
+
+	zones := make(map[string]*v1alpha1.Zone)
+
+	for nodeNum, resList := range perNuma {
+		zone := &v1alpha1.Zone{
+			Type:      "Node",
+			Resources: make(v1alpha1.ResourceInfoMap),
+		}
+
+		costs, err := makeCostsPerNumaNode(nodeResourceData.numaInfo, nodeResourceData.numaDists, nodeNum)
+		if err != nil {
+			// TODO log and move forward
+		} else {
+			zone.Costs = costs
+		}
+
+		for name, resData := range resList {
+			allocatableQty := *resource.NewQuantity(resData.allocatable, resource.DecimalSI)
+			capacityQty := *resource.NewQuantity(resData.capacity, resource.DecimalSI)
+			zone.Resources[name.String()] = v1alpha1.ResourceInfo{
+				Allocatable: allocatableQty.String(),
+				Capacity:    capacityQty.String(),
+			}
+		}
+		zones[makeZoneName(nodeNum)] = zone
+	}
+	return zones
+}
+
+func GetContainerDevicesFromAllocatableResources(availRes *podresourcesapi.AvailableResourcesResponse, cpus *cpus.CPUs) []*podresourcesapi.ContainerDevices {
+	var contDevs []*podresourcesapi.ContainerDevices
+	for _, dev := range availRes.GetDevices() {
+		contDevs = append(contDevs, dev)
+	}
+
+	cpuRes := make(map[int][]string)
+	for _, cpuID := range availRes.GetCpuIds() {
+		nodeID, ok := cpus.GetNodeIDForCPU(int(cpuID))
+		if !ok {
+			// TODO log this is weird
+			continue
+		}
+
+		cpusPerNuma := cpuRes[nodeID]
+		cpusPerNuma = append(cpusPerNuma, fmt.Sprintf("%d", cpuID))
+		cpuRes[nodeID] = cpusPerNuma
+	}
+
+	for nodeID, cpuList := range cpuRes {
+		contDevs = append(contDevs, &podresourcesapi.ContainerDevices{
+			ResourceName: string(v1.ResourceCPU),
+			DeviceIds:    cpuList,
+			Topology: &podresourcesapi.TopologyInfo{
+				Nodes: []*podresourcesapi.NUMANode{
+					{ID: int64(nodeID)},
+				},
+			},
+		})
+	}
+
+	return contDevs
+}
+
```

Comment:
Lets add a comment providing a brief description of what this function does

Reply:
sure, done.